myhomework2.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+
+// не звертайте на цю функцію уваги
+// вона потрібна для того щоб правильно зчитати вхідні дані
+function readHttpLikeInput()
+{
+    $f = fopen('php://stdin', 'r');
+    $store = "";
+    $toread = 0;
+    while ($line = fgets($f)) {
+        $store .= preg_replace("/\r/", "", $line);
+        if (preg_match('/Content-Length: (\d+)/', $line, $m))
+            $toread = $m[1] * 1;
+        if ($line == "\r\n")
+            break;
+    }
+    if ($toread > 0)
+        $store .= fread($f, $toread);
+    return $store;
+}
+
+$contents = readHttpLikeInput();
+
+function parseTcpStringAsHttpRequest($string)
+{
+    return array(
+        "method" => getMethod($string),
+        "uri" => getUri($string),
+        "headers" => getHeaders($string),
+        "body" => getBody($string),
+    );
+}
+
+function getHeaders($string)
+{
+    $lines = explode("\n", $string);
+
+    $headers = array();
+
+    for ($i = 1; $i < sizeof($lines); $i++) {
+        if (strpos($lines[$i], ":")) {
```

Comment:
use str_contains()

Reply:
you will get wrong behavior with a case when `$lines[$i] === ': header'`

Reply:
i replaced strpos with str_contains but I don't understand how to solve this case "$lines[$i] === ': header'"